### ISSUE TITLE
[1.15.2] Can't remove items from RS grid when connected to DimStorage chest

### ISSUE THREAD

**Issue by ocram9** [refinedmods/refinedstorage#2568]
#### Issue description:
When I connect a dimstorage chest to my RS system using the external storage I can't take any items from my RS grid. Not just the items from the dimstorage chest. I already reported it on the dimstorage github but they told me I should contact you: https://github.com/Edivad99/DimStorage/issues/27

#### Steps to reproduce:
1. Connect a dimstorage chest to an RS system using external storage. 
2. Try to remove items from the RS system (not only the items from the chest but also from drives). If you're fast enough you might be able to take something once but not after that.

#### Versions:

- Minecraft: 1.15.2
- Forge: 31.2.9
- Refined Storage: 1.8.4
- DimStorage: 3.1.5

Does this issue occur on a server? [yes/no]
Yes

**Comment by Darkere** [comment]
Commented on the issue over at their github as the issue is at least 90% their fault.

I'm not certain how and if we can improve how we react to a NEIHBORS_CHANGED event.

**Comment by Darkere** [comment]
Some more looking at our side we do have a problem here. 

We are currently reacting to any NEIGHBORS_CHANGED (this happens lots, for example when a player opens a chest) with invalidating the Storage cache. This was fine in 1.12, but in 1.15 this invalidation gives each stack a new UUID which in turn causes them to not be extractable from a grid. 

Not seeing too many good solutions. Maybe we can call something else if the block state didn't change?

**Comment by GsakuL** [comment]
Not sure, if it helps, but this also happens with Ender Chests from Ender Storage

Versions:
- (Pack: Valhelsia 2 - 2.2.5a)
- Minecraft: 1.15.2
- Forge: 31.2.31
- Refined Storage: 1.8.8
- Ender Storage: 2.5.0.143

**Comment by AdamNisbett** [comment]
For what it's worth, I also saw this issue when trying to connect with the Dank Storage mod.

Pack: Enigmatica 6 0.2.13
Minecraft 1.16.1
Forge 32.0.108
Refined Storage 1.9.2
refinedstorageaddons-0.7

**Comment by Famous5000** [comment]
Sorry to necro, but this issue is still present on ATM6:TTS. Using this with Compact Drawers also causes the system to spaz out when an item is added to a drawer. This can be ignored if you use the drawer controller instead of the drawer itself when using the External Storage. Pack version 1.1.6, will get mod versions if this helps.

**Comment by raoulvdberge** [comment]
Yes, we know the issue is still present, that's why the bug is still open.
